integration/pages/BasePage.js

```diff
@@ -0,0 +1,52 @@
+const config = require('../utils/config.js');
+const until = require('selenium-webdriver/lib/until');
+
+class BasePage {
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async visit(route) {
+    await this.driver.get(config.baseUrl + route);
+  }
+
+  find(locator) {
+    return this.driver.findElement(locator);
+  }
+
+  async findAll(locator) {
+    return await this.driver.findElements(locator);
+  }
+
+  async click(locator) {
+    await this.find(locator).click();
+  }
+
+  async type(locator, inputText) {
+    await this.find(locator).sendKeys(inputText);
+  }
+
+  async isDisplayed(locator, timeout) {
+    if (timeout) {
+      await this.driver.wait(until.elementLocated(locator), timeout);
```

Comment:
Do we need this or can we just check for elementIsVisible?

Reply:
@LeviSklaroff Maybe we could use this instead of writing our own loop :D

Reply:
> Need to include GitHub action and an actual test :)

These will be in separate PRs not this one. Vidya told me to break down tasks and PRs.

Reply:
But i mean can we just wait for it to be visible which then should include that it exists?

Reply:
With `timeout` we actually wait until it exists.

> can elementIsVIsible mean it is located?

No. The `elementIsVisible(WebElement)` need a `WebElement` input. We have to first `find` (locate) this element. This is why we need `await this.driver.wait(until.elementLocated(locator), timeout)` to first check if the element is able to located or not.

Reply:
The alternative solution could be: In a previous test we already check  the element exists, and then we just need to check if it is displayed.